optimum/exporters/tflite/convert.py

```diff
@@ -132,10 +138,180 @@ def validate_model_outputs(
+                    inputs_to_keep = args_to_keep | kwargs_to_keep
+                yield sig_name, {name: value for name, value in example.items() if name in inputs_to_keep}
+
+    return representative_dataset
+
+
+def prepare_converter_for_quantization(
+    model: "TFPreTrainedModel",
+    config: "TFLiteConfig",
+    preprocessor: Optional[Preprocessor],
+    signatures: Dict[str, Callable],
+    quantization_config: "QuantizationConfig",
+    converter: "tf.lite.TFLiteConverter",
+):
+    import tensorflow as tf
+
+    if (
+        not config.supports_quantization_approach(quantization_config.approach)
+        and not quantization_config.fallback_to_float
+    ):
+        raise QuantizationApproachNotSupported(
+            f"{model.config.model_type} do not support full {quantization_config.approach} quantization, use "
+            "fallback_to_float=True to fallback to the float implementation for the unsupported ops."
+        )
+
+    str_to_dtype = {"int8": tf.int8, "uint8": tf.uint8}
+    if quantization_config.approach in [QuantizationApproach.INT8, QuantizationApproach.INT8x16]:
+        if preprocessor is None:
+            raise ValueError(
+                "A preprocessor must be passed for INT8 and INT8x16 quantization since it is needed to preprocess "
+                "the calibration dataset."
+            )
+
+        converter.optimizations = [tf.lite.Optimize.DEFAULT]
+
+        # Handling the calibration dataset:
+        # - Either loading the default dataset if no calibration dataset was provided or the required dataset,
+        # - Splitting the dataset with the provided a dataset split or with the first split if none is provided.
+        # - Shuffling the split.
+        # - Selecting num_calibration_samples in the dataset split.
+        # - Batching the dataset.
+        # - Converting it to the TensorFlow format.
+        from ...exporters import TasksManager
+
+        task = TasksManager.get_task_from_model(model)
+        preprocessor_kwargs = {}
+        if isinstance(preprocessor, PreTrainedTokenizerBase):
+            preprocessor_kwargs["max_length"] = config.sequence_length
+        task_processor = TaskProcessorsManager.get_task_processor_class_for_task(task)(
+            model.config, preprocessor, preprocessor_kwargs
+        )
+
+        if task == "token-classification" and model.config.model_type in {
+            "bloom",
+            "camembert",
+            "deberta",
+            "gpt2",
+            "roberta",
+        }:
+            preprocessor.add_prefix_space = True
+
+        load_smallest_split = quantization_config.calibration_split is None
+        if load_smallest_split:
+            logger.warning(
+                "Since no calibration split was provided for the calibration dataset, the smallest split will be "
+                "used if the dataset contains multiple splits."
+            )
```

Comment:
We are not validating yet. When we are validating I will make sure it will not be the case.
Without this, the user would need to pass the split for a "default" dataset (when none is provided, we use a default one), which is annoying.

Reply:
Yes, I think it would be better if we force the user to pass a split if e.g. the `train` split is not in the dataset. Or maybe at least log which split is auto-selected (maybe it is already done)?

Reply:
Added a log.